test/unit/test_telemetry.py

```diff
@@ -132,3 +134,44 @@ def test_telemetry_send_batch_disabled():
     client.send_batch()
     assert client.buffer_size() == 1
     assert rest_call.call_count == 0
+
+
+def test_generate_telemetry_with_driver_info():
+    telemetry_data = generate_telemetry_data()
+    assert (
+        len(telemetry_data) == 2
+        and telemetry_data[TelemetryField.KEY_DRIVER_TYPE.value] == CLIENT_NAME
+        and telemetry_data[TelemetryField.KEY_DRIVER_VERSION.value]
+        == SNOWFLAKE_CONNECTOR_VERSION
+    )
```

Comment:
Bit more concise, length is tested implicitly:
```suggestion
        assert generate_telemetry_data() == {
            TelemetryField.KEY_DRIVER_TYPE.value: CLIENT_NAME,
            TelemetryField.KEY_DRIVER_VERSION.value: SNOWFLAKE_CONNECTOR_VERSION,
        }
```

This could be done to the rest of the test.

Reply:
updated